application/client/src/app/ui/layout/workspace/no-tabs-content/component.ts

```diff
@@ -53,32 +54,47 @@ export class LayoutWorkspaceNoContent {
             .then((files: File[]) => {
                 if (files.length === 0) {
                     return;
+                } else if (files.length === 1) {
+                    files.forEach((file: File) => {
+                        switch (file.type) {
+                            case FileType.Any:
+                            case FileType.Text:
+                                this.ilc()
+                                    .services.system.opener.file(file)
+                                    .text()
+                                    .catch((err: Error) => {
+                                        this.log().error(
+                                            `Fail to open text file; error: ${err.message}`,
+                                        );
+                                    });
+                                break;
+                            case FileType.Dlt:
+                                this.ilc()
+                                    .services.system.opener.file(file)
+                                    .dlt()
+                                    .catch((err: Error) => {
+                                        this.log().error(
+                                            `Fail to open dlt file; error: ${err.message}`,
+                                        );
+                                    });
+                                break;
+                        }
+                    });
+                } else {
+                    this.ilc()
+                        .services.system.session.getTabsService()
+                        .add({
+                            name: 'Multiple Files',
+                            active: true,
+                            closable: true,
```

Comment:
use more high level API:
```
this.ilc().services.system.session.add().tab(
                      {
                            name: 'Multiple Files',
                            active: true,
                            closable: true,
                            content: {
                                factory: TabSourceMultipleFiles,
                                inputs: {
                                    files: files,
                                },
                       }
);
```
it will give you more `inputs` (including `TabControls`) in your component